plugins/bazaar/src/components/SortView/SortView.tsx

```diff
@@ -0,0 +1,163 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { useState } from 'react';
+import {
+  Content,
+  ContentHeader,
+  SupportButton,
+  Progress,
+} from '@backstage/core-components';
+import { AddProjectDialog } from '../AddProjectDialog';
+import { AlertBanner } from '../AlertBanner';
+import { ProjectPreview } from '../ProjectPreview/ProjectPreview';
+import { Button, makeStyles, Link } from '@material-ui/core';
+import { useAsync } from 'react-use';
+import {
+  Entity,
+  EntityRef,
+  stringifyEntityRef,
+} from '@backstage/catalog-model';
+import { useApi } from '@backstage/core-plugin-api';
+import {
+  catalogApiRef,
+  CATALOG_FILTER_EXISTS,
+} from '@backstage/plugin-catalog-react';
+import { BazaarProject } from '../../types';
+import { bazaarApiRef } from '../../api';
+
+const useStyles = makeStyles({
+  container: {
+    marginTop: '2rem',
+  },
+});
+
+export const SortView = () => {
+  const classes = useStyles();
+  const [openAdd, setOpenAdd] = useState(false);
+  const [openNoProjects, setOpenNoProjects] = useState(false);
+  const [catalogEntities, setCatalogEntities] = useState<Entity[]>([]);
+  const [bazaarMembers, setBazaarMembers] = useState<Map<EntityRef, number>>(
+    new Map(),
+  );
+  const [bazaarProjects, setBazaarProjects] = useState<BazaarProject[]>([]);
+  const bazaarApi = useApi(bazaarApiRef);
+  const catalogApi = useApi(catalogApiRef);
+
+  const compareProjectsByDate = (
+    a: BazaarProject,
+    b: BazaarProject,
+  ): number => {
+    const dateA = new Date(a.updatedAt).getTime();
+    const dateB = new Date(b.updatedAt).getTime();
+    return dateB - dateA;
+  };
+
+  const handleCloseNoProjects = () => {
+    setOpenNoProjects(false);
+  };
+
+  const { loading } = useAsync(async () => {
+    const entities = await catalogApi.getEntities({
+      filter: {
+        kind: 'Component',
+        'metadata.annotations.backstage.io/edit-url': CATALOG_FILTER_EXISTS,
+      },
+      fields: ['apiVersion', 'kind', 'metadata', 'spec'],
+    });
+
+    const response = await bazaarApi.getEntities();
+    const dbProjects: BazaarProject[] = [];
+    const bazaarProjectRefs: string[] = [];
+
+    response.data.forEach((project: any) => {
+      dbProjects.push({
+        entityRef: project.entity_ref,
+        name: project.name,
+        status: project.status,
+        announcement: project.announcement,
+        community: project.community,
+        updatedAt: project.updated_at,
+      });
+
+      bazaarProjectRefs.push(project.entity_ref);
+    });
+
+    setBazaarMembers(await bazaarApi.getMemberCounts(dbProjects));
```

Comment:
Getting this directly from the API through the database response when listing bazaar items would probably be beneficial. Like, the backend doing a fancy JOIN with the members table etc in order to get the counts upfront or so. `{ id: 123, status: 'proposed', membersCount: 2, ... }` But that could be left as a future improvement!